### ISSUE TITLE
stack script needs network to work normally?

### ISSUE THREAD

**Issue by ppzzppz** [commercialhaskell/stack#5103]
The following file:
```haskell
#!/usr/bin/env stack
{- stack --resolver global runghc
 --package process
 -}
{-# LANGUAGE OverloadedStrings #-}

main = do
  putStrLn "hello"
```
run command : stack xx.hs

Error reported in case of  without network
```
HttpExceptionRequest Request {
  host                 = "mirrors.tuna.tsinghua.edu.cn"
  port                 = 80
  secure               = False
  requestHeaders       = [("Accept","application/json"),("User-Agent","The Haskell Stack")]
  path                 = "/stackage/snapshots.json"
  queryString          = ""
  method               = "GET"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
 (ConnectionFailure Network.Socket.getAddrInfo (called with preferred socket type/protocol: AddrInfo {addrFlags = [AI_ADDRCONFIG], addrFamily = AF_UNSPEC, addrSocketType = Stream, addrProtocol = 0, addrAddress = <assumed to be undefined>, addrCanonName = <assumed to be undefined>}, host name: Just "mirrors.tuna.tsinghua.edu.cn", service name: Just "80"): does not exist (nodename nor servname provided, or not known))
```
expect： to work without network connection

**Comment by qrilka** [comment]
@ppzzppz this seems to be an bug with an easy fix if the point is about the `global` resolver, see https://github.com/commercialhaskell/stack/blob/stable/src/Stack/Config.hs#L153 - snapshots are not used for the `ARGlobal` value.
PR with a fix will be welcomed.

**Comment by ppzzppz** [comment]
I have tried and find that only “resolver global” doesn’t meet our expectation. The problem doesn’t seem too difficult, and I will try to fix

**Comment by qrilka** [comment]
Much appreciated, let me know if I could help with it

**Comment by snoyberg** [comment]
Resolved by #5104, thanks!
